### PR TITLE
Fix operating flow rate when Source Side Design Volume Flow Rate is autosized for an air-source HeatPump:PlantLoop:EIR:* object

### LINKED ISSUE #8755: Air-Source HeatPump:PlantLoop:EIR:Heating and Cooling not transferring energy if capacity is Autosized
Issue overview
--------------
This issue was discovered when running a model using HeatPump:PlantLoop:EIR:Heating and HeatPump:PlantLoop:EIR:Cooling objects with the Load Side Design Volume Flow Rate hard-sized but the Source Side Design Volume Flow Rate and Reference Capacity autosized (example object shown below).

```
HeatPump:PlantLoop:EIR:Cooling,
  CoolSys1 Chiller 1,            !- Name
  CoolSys1 Chiller 1 Inlet Node, !- Load Side Inlet Node Name
  CoolSys1 Supply Equipment Outlet Node 1,  !- Load Side Outlet Node Name
  AirSource,               !- Condenser Type
  CoolSys1 OA Node,  !- Source Side Inlet Node Name
  Outdoor Air Outlet Node, !- Source Side Outlet Node Name
  HeatSys1 Boiler 1,            !- Companion Heating Heat Pump Name
  0.009,                   !- Load Side Design Volume Flow Rate {m3/s}
  autosize,                      !- Source Side Design Volume Flow Rate {m3/s}
  autosize,                  !- Reference Capacity
  5.69,                     !- Reference COP
  ,                        !- Sizing Factor
  NXNCoolingCapCurveFuncOAT,       !- Heating Capacity Modifier Function of Temperature Curve Name
  NXN_CoolingEIRCurveFuncOAT,       !- Electric Input to Heating Output Ratio Modifier Function of Temperature Curve Name
  NXN_PLE_EIR;        !- Electric Input to Heating Output Ratio Modifier Function of Part Load Ratio Curve Name

```

In running the model, the output results show no Heat Pump Load Side Heat Transfer Energy to attempt to satisfy the heating or cooling demand on either loop. 
image
 
The capacities of the heat pumps are sized to reasonable values during the load side sizing calculation but it appears that during the [source side sizing calculation](https://github.com/NREL/EnergyPlus/blob/b622e27ca3c96566d6bbc8c6429ae971f880d445/src/EnergyPlus/PlantLoopHeatPumpEIR.cc#L875), the `tmpCapacity` value of -99999 is used to calculate the `designSourceSideHeatTransfer` constant, which is then used to calculate `tmpSourceVolFlow` which ends up being the value of `this->sourceSideDesignVolFlowRate`, resulting in a negative number.  This is passed on to calculate 
this->sourceSideDesignMassFlowRate for future simulation calculations and results in no flow and thus no heat transfer.  Below is the if statement where the `tmpSourceVolFlow` is calculated in this situation.

```
    if (this->sourceSideDesignVolFlowRateWasAutoSized) {
        // load-side capacity should already be set, so unless the flow rate is specified, we can set
        // an assumed reasonable flow rate since this doesn't affect downstream components
        Real64 DeltaT_src = 10;
        // to get the source flow, we first must calculate the required heat impact on the source side
        // First the definition of COP: COP = Qload/Power, therefore Power = Qload/COP
        // Then the energy balance:     Qsrc = Qload + Power
        // Substituting for Power:      Qsrc = Qload + Qload/COP, therefore Qsrc = Qload (1 + 1/COP)
        Real64 const designSourceSideHeatTransfer = tmpCapacity * (1 + 1 / this->referenceCOP);
        // To get the design source flow rate, just apply the sensible heat rate equation:
        //                              Qsrc = rho_src * Vdot_src * Cp_src * DeltaT_src
        //                              Vdot_src = Q_src / (rho_src * Cp_src * DeltaT_src)
        tmpSourceVolFlow = designSourceSideHeatTransfer / (rhoSrc * CpSrc * DeltaT_src);
    } else if (!this->sourceSideDesignVolFlowRateWasAutoSized && this->sourceSideDesignVolFlowRate > 0) {

```

I've also included the defect file used below:

[in.zip](https://github.com/NREL/EnergyPlus/files/6430684/in.zip)

### Details
Some additional details for this issue (if relevant):
 - EnergyPlus v9.5
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by matthew-larson**
Pull request overview
---------------------
 - Fixes #8755 
 - This pull request corrects an issue where the `this->loadSideMassFlowRate` and `this->sourceSideMassFlowRate` were being set to 0 when the air-source heat pump Source Side Design Volume Flow Rate was set to 'Autosize'.  When autosized, `this->sourceSideMassFlowRate` resulted in a negative value, which was turning off the entire heat pump.  This portion of the if statement was removed to prevent this from occurring by only looking at `this->loadSideMassFlowRate` since `this->sourceSideMassFlowRate` is not used when an air-source heat pump is specified.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally

### CODE REVIEW

**Comment by Myoldmopar** [comment]
CI looks happy here.  I'll get it building locally and look over the changes.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 1]
```diff
@@ -271,7 +271,7 @@ void EIRPlantLoopHeatPump::setOperatingFlowRatesASHP(EnergyPlusData &state)
                                              this->loadSideLocation.compNum);
 
         // if there's no flow in one, try to turn the entire heat pump off
-        if (this->loadSideMassFlowRate <= 0.0 || this->sourceSideMassFlowRate <= 0.0) {
+        if (this->loadSideMassFlowRate <= 0.0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:274]: I was initially surprised by this change, but I see the function scope is only for ASHP, so this is good to go.

**Review by Myoldmopar — APPROVED** [review]
No problem from me on this one.  Building and testing locally.

**Comment by Myoldmopar** [comment]
Verified the fix locally, seeing the heat pump not do anything in develop but run in this branch.  Merged up with develop and built and tested full unit test suite too, everything is good.  Merging.  Thanks @matthew-larson!

**Comment by lincolnharmer** [comment]
Has anyone looked into why the this->sourceSideMassFlowRate is going negative when the object is autosized? This fix for ASHP might be a band-aid and the root-cause still exists. This fix can't be used for the waterSource version of this object because you will get a divide by zero exception when trying to calculate this->sourceSideOutletTemp.

**Pull request merged**